google/cloud/future_generic.h

```diff
@@ -60,6 +60,15 @@ class future final : private internal::future_base<T> {
   // NOLINTNEXTLINE(google-explicit-constructor)
   future(future<future<T>>&& rhs) noexcept(false);
 
+  /**
+   * Creates a future from a future whose result type is convertible to this
+   * future's result type.
+   */
+  template <class U,
+            typename Enable = std::enable_if<std::is_constructible<T, U>{}>>
```

Comment:
This is allowing code to compile when it shouldn't: https://godbolt.org/z/zPzefx6GG
Note the values of `ShouldBe0` and `ShouldBe1`.  You want  `is_constructible<>::value` and you want `std::enable_if<>::type`.

Reply:
Good catch, will fix. Do you have the ability to write nc tests?

Reply:
Not at this time.  And I am not sure how we would write them.  But you should be able to write a `static_assert()` that verifies SFINAE works as expected?

Reply:
Done.